src/video.ts

```diff
@@ -0,0 +1,42 @@
+import { BlockElement } from "@guardian/content-api-models/v1/blockElement";
+import { Option, None, Some } from 'types/option';
+import { Atoms } from "@guardian/content-api-models/v1/atoms";
+
+interface Video {
+    posterUrl: string;
+    videoId: string;
+    duration?: number;
+}
+
+const parseVideo = (element: BlockElement, atoms?: Atoms): Option<Video> => {
+    if (!atoms) {
+        return new None();
+    }
+
+    const id = element.contentAtomTypeData?.atomId
+    const atom = atoms.media?.find(media => media.id === id);
+
+    if (atom?.data?.kind !== "media") {
+        return new None();
+    }
+
+    const { posterUrl, duration, assets, activeVersion } = atom?.data?.media;
+    const videoId = assets
```

Comment:
This logic looks very similar to the one in `atoms.ts`, is this on purpose?

Reply:
Yes it is just parsing the atom data from the `Content` type. Maybe it could be shared, there's two places where we use the atom, in the article body and the main media section at the top.

Reply:
Oh I see, nah it's probably not worth trying to make it generic then, to different use cases that happen to be similar